Objective-C/CBLReplicator.mm

```diff
@@ -752,10 +752,11 @@ - (bool) filterDocument: (C4String)docID
 // TODO: Remove https://issues.couchbase.com/browse/CBL-3206
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-    auto doc = [[CBLDocument alloc] initWithDatabase: _config.database
-                                          documentID: slice2string(docID)
-                                          revisionID: slice2string(revID)
-                                                body: body];
+    CBLCollection* c = [_config.database defaultCollectionOrThrow];
```

Comment:
Pass collectionSpec to this method. 

It's OK for now to get the default collection. Please create a ticket to make the replicator support collection.

Reply:
https://issues.couchbase.com/browse/CBL-3370